modules/cucumber-reportportal-formatter.js

```diff
@@ -15,943 +15,43 @@
  */
 
 const ReportPortalClient = require('@reportportal/client-javascript');
-const Table = require('cli-table3');
 const utils = require('./utils');
-const Context = require('./context');
-const DocumentStorage = require('./documents-storage');
 const pjson = require('../package.json');
-const itemFinders = require('./itemFinders');
-const {
-  AFTER_HOOK_URI_TO_SKIP,
-  RP_ENTITY_LAUNCH,
-  STATUSES,
-  LOG_LEVELS,
-  CUCUMBER_EVENTS,
-  RP_EVENTS,
-  TABLE_CONFIG,
-  CUCUMBER_MESSAGES,
-} = require('./constants');
-const Storage = require('./storage');
 
 const createRPFormatterClass = (config) => {
   let Formatter;
+  let module;
   try {
+    // eslint-disable-next-line global-require
+    module = require('./api/new');
     // eslint-disable-next-line global-require
     Formatter = require('@cucumber/cucumber').Formatter;
   } catch (e) {
     // eslint-disable-next-line global-require
     Formatter = require('cucumber').Formatter;
+    // eslint-disable-next-line global-require
+    module = require('./api/old');
```

Comment:
I think better to rename it to `deprecated` as it works with deprecated cucumber's versions.
The `new` api can be renamed accordingly to `actual` or `current` or smth like that.